src/util.ts

```diff
@@ -432,10 +458,14 @@ export async function fetchCalldataDecoder (_data: Uint8Array | string, to: stri
     // Convert the chainId to a number and use it to determine if we can call out to
     // an etherscan-like explorer for richer data.
     const chainId = Number(_chainId);
-    const supportedChain = NETWORKS_BY_CHAIN_ID[chainId];
+    const cachedNetwork = NETWORKS_BY_CHAIN_ID[chainId];
+    const supportedChain = cachedNetwork
+      ? cachedNetwork
+      : await fetchExternalNetworkForChainId(chainId);
     if (supportedChain) {
       const baseUrl = supportedChain.baseUrl;
-      const url = `${baseUrl}/api?module=contract&action=getabi&address=${to}&apiKey=${process.env.ETHERSCAN_KEY}`
```

Comment:
Oooh we don't want to lose that API key thing or testing will be painfully slow. Should add the `apiKey` arg conditional on `process.env.ETHERSCAN_KEY` existing

Reply:
Ehhh not to be too nitpicky but I would prefer you conditionally add the argument rather than pass `apiKey=undefined` to the url